Mathlib/RingTheory/Polynomial/Opposites.lean

```diff
@@ -33,93 +33,91 @@ def opRingEquiv (R : Type _) [Semiring R] : R[X]ᵐᵒᵖ ≃+* Rᵐᵒᵖ[X] :=
   ((toFinsuppIso R).op.trans AddMonoidAlgebra.opRingEquiv).trans (toFinsuppIso _).symm
 #align polynomial.op_ring_equiv Polynomial.opRingEquiv
 
-/-!  Lemmas to get started, using `op_ring_equiv R` on the various expressions of
-`finsupp.single`: `monomial`, `C a`, `X`, `C a * X ^ n`. -/
+/-!  Lemmas to get started, using `opRingEquiv R` on the various expressions of
+`Finsupp.single`: `monomial`, `C a`, `X`, `C a * X ^ n`. -/
 
 
--- for maintenance purposes: `by simp [op_ring_equiv]` proves this lemma
 @[simp]
 theorem opRingEquiv_op_monomial (n : ℕ) (r : R) :
-    opRingEquiv R (op (monomial n r : R[X])) = monomial n (op r) := by
-  simp only [op_ring_equiv, RingEquiv.trans_apply, RingEquiv.op_apply_apply,
-    RingEquiv.toAddEquiv_eq_coe, AddEquiv.mulOp_apply, [anonymous], AddEquiv.coe_trans,
-    op_add_equiv_apply, RingEquiv.coe_toAddEquiv, op_add_equiv_symm_apply, Function.comp_apply,
-    unop_op, to_finsupp_iso_apply, to_finsupp_monomial, AddMonoidAlgebra.opRingEquiv_single,
-    to_finsupp_iso_symm_apply, of_finsupp_single]
+    opRingEquiv R (op (monomial n r : R[X])) = monomial n (op r) := by simp [opRingEquiv]
```

Comment:
Ideally we'd keep the squeezed proof, as its faster

Reply:
Is non-squeezed proof still slow in Lean 4? It's possible that Lean 4 `simp` is faster than Lean 3 `simp` here. How do we measure this in Lean 4?

Reply:
Otherwise LGTM.

Reply:
> Ideally we'd keep the squeezed proof, as its faster

The squeezed proof from mathlib3 is unsalvageable. `ring_equiv.op_apply_apply` has no direct mathlib4 equivalent, as can be seen by putting `@[simps?]` before the generator of `op_apply_apply` in both mathlibs:

```
[simps] > adding projection ring_equiv.op_apply_apply:
        > ∀ {α : Type u_1} {β : Type u_2} [_inst_1 : has_add α] [_inst_2 : has_mul α] [_inst_3 : has_add β]
[_inst_4 : has_mul β] (f : α ≃+* β) (ᾰ : αᵐᵒᵖ),
  ⇑(⇑ring_equiv.op f) ᾰ = (⇑add_equiv.mul_op f.to_add_equiv).to_fun ᾰ
```

The closest mathlib4 equivalent is `RingEquiv.op_apply_apply_unop` but its definition is not the same:
```
[simps.verbose] adding projection RingEquiv.op_apply_apply_unop:
      ∀ {α : Type u_1} {β : Type u_2} [inst : Add α] [inst_1 : Mul α] [inst_2 : Add β] [inst_3 : Mul β] (f : α ≃+* β)
        (a : αᵐᵒᵖ), (↑(↑RingEquiv.op f) a).unop = ↑f a.unop
```

> Is non-squeezed proof still slow in Lean 4? It's possible that Lean 4 `simp` is faster than Lean 3 `simp` here. How do we measure this in Lean 4?

It's definitely not slow for me.

Reply:
Since `simp` works in Lean 4, you can do `simp?` in Lean 4 instead of forward-porting Lean 3 squeezed `simp`.

Reply:
> Since `simp` works in Lean 4, you can do `simp?` in Lean 4 instead of forward-porting Lean 3 squeezed `simp`.

To get the closest equivalent lemma I did exactly `@[simps?]` on the Lean 4 lemma:
https://github.com/leanprover-community/mathlib4/blob/master/Mathlib/Algebra/Ring/Equiv.lean#L361

But of course doing the same on `opRingEquiv_op_monomia` returns an error:
```
Invalid `simps` attribute. Target Eq is not a structure
```

What are you asking me to do?

Reply:
@urkud right, that worked; commit pushed. Thanks.